kuma/wiki/forms.py

```diff
@@ -356,14 +392,77 @@ def clean_current_rev(self):
             # If there's no document yet, just bail.
             return current_rev
 
-    def save(self, request, document, **kwargs):
+    def akismet_enabled(self):
+        """
+        Makes sure that users that have been granted the
+        'wiki_akismet_exempted' waffle flag are exempted from spam checks.
+        """
+        api_ready = super(RevisionForm, self).akismet_enabled()
+        user_exempted = waffle.flag_is_active(self.request,
+                                              SPAM_EXEMPTED_FLAG)
+        return api_ready and not user_exempted
+
+    @property
+    def ERROR_MESSAGE(self):
+        return mark_safe(render_to_string('wiki/includes/spam_error.html', {}))
+
+    def akismet_error(self):
+        """
+        Upon errors from the Akismet API records the user, document
+        and date of the attempt for further analysis. Then call the
+        parent class' error handler.
+        """
+        try:
+            document = self.instance.document
+        except ObjectDoesNotExist:
+            document = None
+        # wrapping this in a try/finally to make sure that even if
+        # creating a spam attempt object fails we call the parent
+        # method that raises a ValidationError
+        try:
+            DocumentSpamAttempt(
+                title=self.cleaned_data['title'],
+                slug=self.cleaned_data['slug'],
+                user=self.request.user,
+                document=document,
+            ).save()
```

Comment:
Why not `DocumentSpamAttempt.objects.create(...)`? I'm asking honestly, not suggesting one over the other.